ci/docker-run.sh

```diff
@@ -41,7 +41,12 @@ ARGS=(
   --rm
 )
 
-if [[ -n $CI ]]; then
+if [[ $BUILDKITE_RETRY_COUNT -ge 3 ]]; then
+  echo "Disabling cache due to many retries"
+  DISABLE_CACHE=1
+fi
+
+if [[ -n $CI && -z $DISABLE_CACHE ]]; then
```

Comment:
I think maybe move the `$DISABLE_CACHE` check to line 53 🤔 (the line `if [[ -n $BUILDKITE ]]; then`) 
just in case we add more CI related code in the future and have a possibly to be disabled by the `DISABLE_CACHE` env